jdaviz/configs/cubeviz/plugins/parsers.py

```diff
@@ -140,13 +140,22 @@ def _return_spectrum_with_correct_units(flux, wcs, metadata, data_type, target_w
             category=UserWarning)
         sc = Spectrum1D(flux=flux, wcs=wcs)
 
-    # TODO: Can the unit be defined in a different keyword, e.g., CUNIT1?
     if target_wave_unit is None and hdulist is not None:
-        cunit_key = 'CUNIT3'
+        found_target = False
         for ext in ('SCI', 'FLUX', 'PRIMARY'):  # In priority order
-            if ext in hdulist and cunit_key in hdulist[ext].header:
-                target_wave_unit = u.Unit(hdulist[ext].header[cunit_key])
+            if found_target:
                 break
+            if ext not in hdulist:
+                continue
+            hdr = hdulist[ext].header
+            # The WCS could be swapped or unswapped.
+            for cunit_num in (3, 1):
+                cunit_key = f"CUNIT{cunit_num}"
+                ctype_key = f"CTYPE{cunit_num}"
+                if not found_target and cunit_key in hdr and 'WAVE' in hdr[ctype_key]:
```

Comment:
with the breaks, is it ever possible for `found_target` to be `True` here?

Reply:
It will never be true on first pass. And once it is true, L146 will catch it and break from the outer loop. Did I miss something?

Reply:
Exactly, so in line `155` it is impossible for `found_target` to be False, since 158 will break out of the inner-loop and 146 out of the outer-loop before 155 gets reevaluated, right?

Reply:
Can't `found_target` be false on L155 if the `cunit_num` that matches is 1 and not 3?

Reply:
yes, sorry, I mis-typed in the latest comment.  I meant that it must always be False, and can never be True (I think).  But I guess the extra check doesn't do any harm, I was just trying to understand the logic.

Reply:
Ah, in that case, yes, I think you are right... 😸